src/Shared/LayoutRenderers/AspNetResponseCookieLayoutRenderer.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Sockets;
+using System.Text;
+using NLog.Config;
+using NLog.LayoutRenderers;
+using NLog.Web.Enums;
+using NLog.Web.Internal;
+#if !ASP_NET_CORE
+using System.Collections.Specialized;
+using System.Web;
+using Cookies = System.Web.HttpCookieCollection;
+#else
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+using Microsoft.AspNetCore.Http;
+#endif
+
+namespace NLog.Web.LayoutRenderers
+{
+    /// <summary>
+    /// ASP.NET Response Cookie
+    /// </summary>
+    /// <example>
+    /// <para>Example usage of ${aspnet-response-cookie}</para>
+    /// <code lang="NLog Layout Renderer">
+    /// ${aspnet-response-cookie:OutputFormat=Flat}
+    /// ${aspnet-response-cookie:OutputFormat=JsonArray}
+    /// ${aspnet-response-cookie:OutputFormat=JsonDictionary}
+    /// ${aspnet-response-cookie:OutputFormat=JsonDictionary:CookieNames=username}
+    /// ${aspnet-response-cookie:OutputFormat=JsonDictionary:Exclude=access_token}
+    /// </code>
+    /// </example>
+    [LayoutRenderer("aspnet-response-cookie")]
+    public class AspNetResponseCookieLayoutRenderer : AspNetLayoutMultiValueRendererBase
+    {
+        /// <summary>
+        /// Cookie names to be rendered.
+        /// If <c>null</c> or empty array, all cookies will be rendered.
+        /// </summary>
+        public List<string> CookieNames { get; set; }
+
+        /// <summary>
+        /// Gets or sets the keys to exclude from the output. If omitted, none are excluded.
+        /// </summary>
+        /// <docgen category='Rendering Options' order='10' />
+#if ASP_NET_CORE
+        public ISet<string> Exclude { get; set; }
+#else
+        public HashSet<string> Exclude { get; set; }
+#endif
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AspNetResponseCookieLayoutRenderer" /> class.
+        /// </summary>
+        public AspNetResponseCookieLayoutRenderer()
+        {
+            Exclude = new HashSet<string>(new[] { "AUTH", "SESS_ID" }, StringComparer.OrdinalIgnoreCase);
+        }
+
+        /// <summary>
+        /// Renders the ASP.NET Cookie appends it to the specified <see cref="StringBuilder" />.
+        /// </summary>
+        /// <param name="builder">The <see cref="StringBuilder" /> to append the rendered data to.</param>
+        /// <param name="logEvent">Logging event.</param>
+        protected override void DoAppend(StringBuilder builder, LogEventInfo logEvent)
+        {
+            var httpResponse = HttpContextAccessor.HttpContext.TryGetResponse();
+            if (httpResponse == null)
+            {
+                return;
+            }
+
+            var cookies = GetCookies(httpResponse);
+#if ASP_NET_CORE
+            if (cookies.Any())
```

Comment:
Can this be removed, so it just becomes `if (cookies.Count > 0)` for all platforms ?

Reply:
Certainly

Reply:
Now that the code-paths are identical. Can the `#if ASP_NET_CORE` be removed around `if (cookies.Count > 0)` ?

Reply:
sorry about that, made the change too quickly